src/main/java/org/testeditor/web/xtext/index/resources/GlobalScopeResourceWithSeparateContextResourceSet.xtend

```diff
@@ -0,0 +1,190 @@
+package org.testeditor.web.xtext.index.resources
+
+import java.io.IOException
+import javax.ws.rs.Consumes
+import javax.ws.rs.POST
+import javax.ws.rs.Path
+import javax.ws.rs.Produces
+import javax.ws.rs.QueryParam
+import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.Response
+import org.eclipse.emf.common.util.URI
+import org.eclipse.emf.ecore.EPackage
+import org.eclipse.emf.ecore.EReference
+import org.eclipse.emf.ecore.resource.Resource
+import org.eclipse.emf.ecore.resource.ResourceSet
+import org.eclipse.emf.ecore.resource.impl.ResourceSetImpl
+import org.eclipse.xtext.resource.impl.ResourceSetBasedResourceDescriptions
+import org.eclipse.xtext.scoping.IGlobalScopeProvider
+import org.eclipse.xtext.util.StringInputStream
+import org.slf4j.LoggerFactory
+
+@Path("/xtext/index/global-scope")
+@Produces(MediaType.APPLICATION_JSON)
+class GlobalScopeResourceWithSeparateContextResourceSet implements GlobalScopeResource {
+
+	protected static val logger = LoggerFactory.getLogger(GlobalScopeResourceWithSeparateContextResourceSet)
+
+	val IGlobalScopeProvider globalScopeProvider
+	val ResourceSetBasedResourceDescriptions index
+
+	new(IGlobalScopeProvider globalScopeProvider, ResourceSetBasedResourceDescriptions index) {
+		this.globalScopeProvider = globalScopeProvider
+		this.index = index
+	}
+
+	@POST // to allow context content to be passed as payload (instead of using query parameters)
+	@Consumes("text/plain")
+	@Produces("application/json")
+	override Response getScope(String context, @QueryParam("contentType") String contentType,
+		@QueryParam("contextURI") String contextURI, @QueryParam("reference") String eReferenceURIString) {
+		try {
+			val eReference = createEReference(eReferenceURIString)
+			val resource = createContextResource(context, contextURI, contentType)
+
+			logger.debug("Delegating to global scope provider='{}'", globalScopeProvider.class.simpleName)
+			val scope = globalScopeProvider.getScope(resource, eReference, null).allElements
+			logger.debug("Global scope provider returned the following elements='{}'", scope.map[name].join(','))
```

Comment:
`scope.map[name].join(',')` might get pretty expensive. I would rather pass `scope.map[name]` to the logger and check if the output still looks okay. Map itself is not terminal and not executed if debug is deactivated. Otherwise I would surround this by a "isDebugEnabled" check